feature-libs/pickup-in-store/components/container/store-list/store-list.component.ts

```diff
@@ -43,16 +54,49 @@ export class StoreListComponent implements OnInit {
       this.productCode
     );
     this.isSearchRunning$ = this.pickupLocationsSearchService.isSearchRunning();
+
+    this.activeCartFacade
+      .getActive()
+      .pipe(
+        tap((cart) => {
+          this.cartId = cart.guid ?? '';
+          this.userId = cart.user?.uid ?? '';
+          cart.entries &&
+            cart.entries.forEach((entry) => {
+              if (entry.product?.code === this.productCode) {
+                this.entryNumber = entry.entryNumber ?? -1;
+                this.quantity = entry.quantity as number;
+              }
+            });
+        })
+      )
+      .subscribe();
   }
 
   onSelectStore(store: PointOfServiceStock) {
     const { stockInfo: _, ...pointOfService } = store;
-    const { name, displayName } = pointOfService;
+    const { name = '', displayName } = pointOfService;
+
+    const payload: SetPickupOptionInStorePayload = {
+      deliveryPointOfService: {
+        name,
+      },
+      quantity: this.quantity,
+    };
+
     this.preferredStoreService.setPreferredStore({ name, displayName });
     this.intendedPickupLocationService.setIntendedLocation(this.productCode, {
       ...pointOfService,
       pickupOption: 'pickup',
     });
+
+    this.pickupLocationsSearchService.setPickupOptionInStore(
+      this.cartId,
+      this.entryNumber,
+      this.userId,
+      payload
+    );
```

Comment:
We don't want to be doing this for items before they have been added to the cart, how does this behave on the PDP?